tasks/build/verify_translations.js

```diff
@@ -0,0 +1,57 @@
+import Promise from 'bluebird';
+
+import fromRoot from '../../src/utils/from_root';
+import KbnServer from '../../src/server/kbn_server';
+import * as i18nVerify from '../../src/core_plugins/i18n/server/tool/i18n_verify_keys';
+import * as i18n from '../../src/core_plugins/i18n/server/i18n/i18n';
+
+module.exports = function (grunt) {
+  grunt.registerTask('_build:verifyTranslations', async function () {
+    const parsePaths = [fromRoot('/src/ui/views/*.jade')];
+    const keyPattern = 'i18n\\(\'(.*)\'\\)';
+    const keyPatternRegEx = new RegExp(keyPattern, 'g');
+
+    const serverConfig = {
+      env: 'production',
+      logging: {
+        silent: true,
+        quiet: true,
+        verbose: false
+      },
+      optimize: {
+        useBundleCache: false,
+        enabled: false
+      },
+      server: {
+        autoListen: false
+      },
+      plugins: {
+        initialize: true,
+        scanDirs: [fromRoot('src/core_plugins')]
+      },
+      uiSettings: {
+        enabled: false
+      }
+    };
+    const kbnServer = new KbnServer(serverConfig);
+    await kbnServer.ready();
+
+    const locales = i18n.getRegisteredTranslationLocales();
+    Promise.all(locales.map(function (locale) {
+      return i18n.getTranslationsForLocale(locale).then(function (translations) {
+        i18nVerify.verifyTranslationKeys(parsePaths, translations, keyPatternRegEx).then(function (keysNotTranslated) {
```

Comment:
For me the confusion comes from the fact that `verifyTranslationKeys` is doing too much. I couldn't explain why that function would take file paths and a regex without looking at the internal code. It just seems more clear to me to separate the key fetching and verification steps.